src/ray/gcs/gcs_server/gcs_resource_manager.h

```diff
@@ -133,6 +139,9 @@ class GcsResourceManager : public rpc::NodeResourceInfoHandler {
       const std::shared_ptr<rpc::PlacementGroupLoad> placement_group_load);
 
  private:
+  /// io context
+  instrumented_io_context &io_context_;
```

Comment:
does it make sense to store this in syncer? i.e. when register receiver, the syncer takes both receiver and an io_context.

I think the confusing part to put it here is which member function of GcsResourceManager should post to io_context_.

Reply:
Ideally, all public member functions need to be put into this io_context_ since in reality, they run in io_context_ right now.

I added comments about this.